File: api/src/main/java/net/impactdev/gts/api/data/translators/DataTranslatorManager.java
package net.impactdev.gts.api.data.translators;

import java.util.Collection;

public interface DataTranslatorManager {

    <T> Collection<DataTranslator<T>> get(Class<T> type);

    <T> void register(Class<T> type, DataTranslator<T> translator);

}

File: api/src/main/kotlin/net/impactdev/gts/api/data/translators/DataTranslatorManager.kt
package net.impactdev.gts.api.data.translators

interface DataTranslatorManager {
    operator fun <T> get(type: Class<T>?): Collection<DataTranslator<T>?>?
    fun <T> register(type: Class<T>?, translator: DataTranslator<T>?)
}